### PR TITLE
sql: remove invalid database privileges 

### LINKED ISSUE #68731: sql: disallow granting postgres incompatible database privileges
We support granting SELECT, INSERT, UPDATE, DELETE on databases for the purpose of inheritance. Now that we have default privileges, these should be disallowed.

This should be done for 22.1, in the meanwhile we have a deprecation warning

Epic CRDB-8948

### PR DESCRIPTION
**Pull request by RichardJCai**
sql: remove invalid database privileges 

Release note (sql change): SELECT, INSERT, DELETE, UPDATE can
no longer be granted/revoked on databases. Previously
SELECT, INSERT, DELETE, UPDATE would be converted to ALTER DEFAULT PRIVILEGES
on GRANTs and were revokable but now they are no longer revokable either.

Resolves https://github.com/cockroachdb/cockroach/issues/68731

### CODE REVIEW

**Comment by RichardJCai** [comment]
This is ready for review but we also have to follow this up with a migration and add some tests / case for backup/restore.

**Review by rafiss — COMMENTED** [review]
what is the concern for backup/restore? i thought none of the saved descriptors could have the invalid privs, since we already were converting them into default privs.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72665#-:-MtP1Mpc0K29nfyAy_Sz:b-pgsyrq)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RichardJCai and @stevendanna)

___
*[-- commits, line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72665#-MtP0p05C2wpj0zuY2MF):*
nit: the commit message and PR title should say "delete" or "remove" instead of "deprecate"

deprecate means they are there but will be going away. in other words, they already were deprecates in v21.2.

**Comment by RichardJCai** [comment]
> what is the concern for backup/restore? i thought none of the saved descriptors could have the invalid privs, since we already were converting them into default privs.
> 
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72665#-:-MtP1Mpc0K29nfyAy_Sz:b-pgsyrq)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RichardJCai and @stevendanna)
> 
> _[-- commits, line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72665#-MtP0p05C2wpj0zuY2MF):_ nit: the commit message and PR title should say "delete" or "remove" instead of "deprecate"
> 
> deprecate means they are there but will be going away. in other words, they already were deprecates in v21.2.

Oh right for backup/restore it might be okay, I'll double check this.

Will change to remove.

**Review by rafiss — APPROVED** [review]
lgtm once the commit message itself is also updated

pretty sure there's no backup concerns thanks to this migration: https://github.com/cockroachdb/cockroach/pull/66495/ (specifically the `u.Privileges &= allowedPrivilegesBits`)

File: pkg/sql/logictest/testdata/logic_test/grant_in_txn [hunk 1]
```diff
@@ -94,7 +94,8 @@ testuser  role_foo   YES
 statement ok
 REVOKE role_foo FROM testuser;
 
-statement ok
+statement okinformation_schema
```
Comment (rafiss) [inline pkg/sql/logictest/testdata/logic_test/grant_in_txn:97]: typo here

**Comment by rafiss** [comment]
ah hm, but i see you left this todo -- so i suppose we still do need a migration
```
// TODO(richardcai): Remove this and the syntax for granting these privileges
//    to databases in 22.1. In 22.1, we should have a long-running migration
//    to convert incompatible privileges into default privileges.
//    See: https://github.com/cockroachdb/cockroach/issues/68731
```

**Comment by RichardJCai** [comment]
TFTR! Gonna add migration in following PR

bors r=rafiss

**Pull request merged**